coins/monero/src/tests/seed.rs

```diff
@@ -163,15 +163,113 @@ fn test_classic_seed() {
+        Seed::from_entropy(SeedType::Classic(vector.language), seed.entropy()).unwrap()
+      );
       assert_eq!(seed, Seed::from_string(seed.to_string()).unwrap());
     }
   }
 }
+
+////// POLYSEED TESTS ///////////////
+///
+///
+#[test]
+fn test_polyseed() {
+  struct Vector {
+    language: polyseed::Language,
+    seed: String,
+    key: String,
+    birtday: u64,
+    formal: bool,
+  }
+
+  let vectors = [
+    Vector {
+      language: polyseed::Language::English,
+      seed: "raven tail swear infant grief assist regular lamp \
+      duck valid someone little harsh puppy airport language"
+        .into(),
+      key: "dd76e7359a0ded37cd0ff0f3c829a5ae01673300000000000000000000000000".into(),
+      birtday: 1638446400,
+      formal: true,
+    },
+    Vector {
+      language: polyseed::Language::English,
+      seed: "raven tail swear infant grief assist regular lamp \
+      duck valid someone little harsh puppy airport language "
+        .into(),
+      key: "dd76e7359a0ded37cd0ff0f3c829a5ae01673300000000000000000000000000".into(),
+      birtday: 1638446400,
+      formal: false,
+    },
+    Vector {
+      language: polyseed::Language::English,
+      seed: "rave tail swea infan grie assi regul lamp \
+      duck vali some litt hars pupp airp langua"
+        .into(),
+      key: "dd76e7359a0ded37cd0ff0f3c829a5ae01673300000000000000000000000000".into(),
+      birtday: 1638446400,
+      formal: false,
+    },
+    Vector {
+      language: polyseed::Language::Spanish,
+      seed: "eje fin parte célebre tabú pestaña lienzo puma \
+      prisión hora regalo lengua existir lápiz lote sonoro"
+        .into(),
+      key: "5a2b02df7db21fcbe6ec6df137d54c7b20fd2b00000000000000000000000000".into(),
+      birtday: 3118651200,
+      formal: true,
+    },
+    Vector {
+      language: polyseed::Language::Spanish,
+      seed: "eje fin parte celebre tabu pestana lienzo puma \
+      prision hora regalo lengua existir lapiz lote sonoro"
+        .into(),
+      key: "5a2b02df7db21fcbe6ec6df137d54c7b20fd2b00000000000000000000000000".into(),
+      birtday: 3118651200,
+      formal: false,
+    },
+    Vector {
+      language: polyseed::Language::Spanish,
+      seed: "eje fin part cele tabu pest lien puma \
+      pris hora rega leng exis lapi lote sono"
+        .into(),
+      key: "5a2b02df7db21fcbe6ec6df137d54c7b20fd2b00000000000000000000000000".into(),
+      birtday: 3118651200,
+      formal: false,
+    },
+  ];
+  let features = 0;
+  let polyseed_time_step = 2630000;
+
+  for v in vectors {
+    // string -> key
+    let s1 = Seed::from_string(Zeroizing::new(v.seed)).unwrap();
+    let key_bytes = Zeroizing::new(hex::decode(v.key).unwrap().try_into().unwrap());
+    let data = s1.polyseed_data();
+    assert_eq!(data.entropy, key_bytes);
+    assert_eq!(data.features, features);
+    assert!(data.birthday <= v.birtday);
+    assert!(data.birthday + polyseed_time_step > v.birtday);
+
+    if v.formal {
```

Comment:
I ran into this when testing trimmed classic seeds. The seed parts should always trim and then expand, where possible, in order to always generate the full string value.